webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/score/Score.scala

```diff
@@ -0,0 +1,251 @@
+}
+case class GlobalScore(value: ScoreValue, message: String, details: List[NoDetailsScore])
+
+object GlobalScoreService {
+  def computeGlobalScore(oldScore: List[NoDetailsScore], scores: List[Score[_]]): GlobalScore = {
+
+    val correctScores = scores.foldRight(oldScore) {
+      case (newScore, acc) =>
+        NoDetailsScore(newScore.name, newScore.value, newScore.message) :: acc.filterNot(_.name == newScore.name)
+    }
+    import ScoreValue._
+    val score         = if (correctScores.exists(_.value == E)) { E }
+    else if (correctScores.exists(_.value == D)) { D }
+    else if (correctScores.exists(_.value == C)) {
+      C
+    } else if (correctScores.exists(_.value == B)) {
+      B
+    } else A
+    GlobalScore(score, s"There is at least a Score with ${score.value}", correctScores)
+  }
+}
+
+trait ScoreEvent
+
+case class InventoryScoreEvent(nodeId: NodeId, inventory: Inventory)                  extends ScoreEvent
+case class ComplianceScoreEvent(nodeId: NodeId, compliancePercent: CompliancePercent) extends ScoreEvent
+
+trait ScoreEventHandler {
+  def handle(event: ScoreEvent): IOResult[List[(NodeId, List[Score[_]])]]
+}
+
+object ComplianceScoreEventHandler extends ScoreEventHandler {
+  def handle(event: ScoreEvent): IOResult[List[(NodeId, List[Score[_]])]] = {
+
+    event match {
+      case ComplianceScoreEvent(n, percent) =>
+        val p     = ComplianceSerializable.fromPercent(percent)
+        val score = if (percent.compliance >= 100) {
+          ComplianceScore(A, "Node is compliant at 100%", p)
+        } else if (percent.compliance >= 75) {
+          ComplianceScore(B, "Node is compliant at least at 75%", p)
+        } else if (percent.compliance >= 50) {
+          ComplianceSerializable.fromPercent(percent)
+          ComplianceScore(C, "Node is compliant at least at 50%", p)
+        } else if (percent.compliance >= 25) {
+          ComplianceScore(D, "Node is compliant at least at 25%", p)
+        } else {
+          ComplianceScore(E, "Node is compliant at less then 25%", p)
+        }
+        ((n, score :: Nil) :: Nil).succeed
+      case _                                => Nil.succeed
+    }
+  }
+}
+
+object InventoryEventHandler extends ScoreEventHandler {
+  def handle(event: ScoreEvent): IOResult[List[(NodeId, List[Score[_]])]] = {
+
+    event match {
+      case InventoryScoreEvent(n, inventory) =>
+        val sum         = inventory.node.softwareUpdates.size
+        val security    = inventory.node.softwareUpdates.filter(_.kind == SoftwareUpdateKind.Security).size
+        val patch       = inventory.node.softwareUpdates.filter(_.kind == SoftwareUpdateKind.None).size
+        val defect      = inventory.node.softwareUpdates.filter(_.kind == SoftwareUpdateKind.Defect).size
+        val enhancement = inventory.node.softwareUpdates.filter(_.kind == SoftwareUpdateKind.Enhancement).size
+        val other       = inventory.node.softwareUpdates.filter { s =>
+          s.kind match {
```

Comment:
it looks like we will want to add these info into `CoreNodeFact` at some point in the future